include/preprocess/max_inscribed_ellipsoid.hpp

```diff
@@ -53,20 +62,22 @@ std::pair<std::pair<MT, VT>, bool> max_inscribed_ellipsoid(MT A, VT b, VT const&
 
     VT const bmAx0 = b - A * x0, ones_m = VT::Ones(m);
 
-    MT Q(m, m), Y(m, m), E2(n, n), YQ(m,m), YA(m, n), G(m,m), T(m,n), ATP(n,m), ATP_A(n,n);
+    MT Q(m, m), E2(n, n), YQ(m,m), G(m,m), T(m,n), ATP(n,m), ATP_A(n,n); 
+    DT Y(m);
+    Custom_MT YA(m, n);
 
     A = (ones_m.cwiseProduct(bmAx0.cwiseInverse())).asDiagonal() * A, b = ones_m;
-
-    MT A_trans = A.transpose();
+    Custom_MT A_trans = A.transpose();
 
     int i = 1;
     while (i <= maxiter) {
 
-        Y = y.asDiagonal();
-        E2.noalias() = (A_trans * Y * A).inverse();
+        Y = y.asDiagonal(); 
+        
+        E2.noalias() = MT(A_trans * Y * A).inverse();
```

Comment:
Since Y is a diagonal matrix, could this operation be further optimized?

Reply:
`Y` is defined to be of type Diagonal Matrix, so Eigen will optimize the multiplication as expected.